### DISCUSSION TITLE
Migration/change database from h2 to mysql/MariaDB within deployed version of EJBCA CE

### DISCUSSION THREAD

**Discussion by merryo11** [Keyfactor/ejbca-ce#343] [General]
Dear All, I have created an EJBCA CE 8.0 environment through docker using KeyFactor image with default h2 DB and created few CAs/End Entity Profiles/Certificate Profiles and issued certificates. Now I want to switch the DB from h2 to mysql or mariadb without losing the data that is I dont want to lose my data from this change and want to have all of my previous data to be imported in new DB (mysql or mariadb) so can any one guide me step wise how can I do this.

Any urgent help shall highly be appreciated.

Current running environment is:
Ubuntu 22/Docker 24/ Keyfactor - EJBCA CE with h2 db

Regards

**Comment by primetomas** [comment]
In EJBCA Enterprise there is a database migration tool. https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/command-line-interfaces#CommandLineInterfaces-Local_Database_CLI

Otherwise it's "plain" database issue. There should be resources to find how to migrate from H2 to MariaDB.

Cheers,
Tomas

**Reply by merryo11** [reply]
I got your point. Now I have one point and that is once I do this DB migration then how can I connect my EJBCA portals to new mysql DB keeping in mind that I did installations using docker

**Reply by primetomas** [reply]
It is documented in the dockerhub documentation how to connect the container to an external MariaDB database.
